modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/test/java/com/liferay/dynamic/data/mapping/form/web/internal/display/context/util/DDMFormAdminActionDropdownItemsProviderTest.java

```diff
@@ -0,0 +1,567 @@
+		Assert.assertEquals("copy", dropdownItem.get("symbolLeft"));
+	}
+
+	private void _assertActionDropdownItemEdit(DropdownItem dropdownItem) {
+		Assert.assertEquals(
+			_INVALID_DDM_FORM_INSTANCE, dropdownItem.get("disabled"));
+		Assert.assertEquals(
+			StringBundler.concat(
+				"http//localhost/test?param_mvcRenderCommandName=/admin",
+				"/edit_form_instance;param_redirect=", _CURRENT_URL,
+				";param_formInstanceId=", _FORM_INSTANCE_ID),
+			dropdownItem.get("href"));
+		Assert.assertEquals("edit", dropdownItem.get("label"));
+		Assert.assertEquals("pencil", dropdownItem.get("symbolLeft"));
+	}
+
+	private void _assertActionDropdownItemExport(DropdownItem dropdownItem) {
+		Assert.assertEquals(
+			_INVALID_DDM_FORM_INSTANCE, dropdownItem.get("disabled"));
+		Assert.assertEquals(
+			StringBundler.concat(
+				"javascript:", _PORTLET_NAMESPACE,
+				"exportFormInstance('http//localhost/test?",
+				"param_formInstanceId=", _FORM_INSTANCE_ID, "');"),
+			dropdownItem.get("href"));
+		Assert.assertEquals("export", dropdownItem.get("label"));
+	}
+
+	private void _assertActionDropdownItemPermissions(
+		DropdownItem dropdownItem) {
+
+		Map<String, Object> dataMap = (Map<String, Object>)dropdownItem.get(
+			"data");
+
+		Assert.assertEquals("permissions", dataMap.get("action"));
+
+		String permissionsFormInstanceURL = String.valueOf(
+			dataMap.get("permissionsFormInstanceURL"));
+
+		Assert.assertThat(
+			permissionsFormInstanceURL,
+			CoreMatchers.containsString(
+				"param_modelResource=" +
+					"com.liferay.dynamic.data.mapping.model.DDMFormInstance;"));
+		Assert.assertThat(
+			permissionsFormInstanceURL,
+			CoreMatchers.containsString(
+				"param_mvcPath=/edit_permissions.jsp;"));
+		Assert.assertThat(
+			permissionsFormInstanceURL,
+			CoreMatchers.containsString(
+				"param_resourcePrimKey=" + _FORM_INSTANCE_ID));
+
+		Assert.assertEquals("true", dataMap.get("useDialog"));
+
+		Assert.assertEquals("permissions", dropdownItem.get("label"));
+	}
+
+	private void _assertActionDropdownItemShare(DropdownItem dropdownItem) {
+		Assert.assertEquals(
+			HashMapBuilder.<String, Object>put(
+				"action", "shareForm"
+			).put(
+				"autocompleteUserURL", _AUTOCOMPLETE_USER_URL
+			).put(
+				"localizedName", JSONUtil.put("en_US", "Test US")
+			).put(
+				"portletNamespace", _PORTLET_NAMESPACE
+			).put(
+				"shareFormInstanceURL", _SHARE_FORM_INSTANCE_URL
+			).put(
+				"url", _PUBLISHED_FORM_URL
+			).build(
+			).toString(),
+			String.valueOf(dropdownItem.get("data")));
+		Assert.assertEquals(
+			_INVALID_DDM_FORM_INSTANCE, dropdownItem.get("disabled"));
+		Assert.assertEquals("share", dropdownItem.get("label"));
+		Assert.assertEquals("share", dropdownItem.get("symbolLeft"));
+	}
+
+	private void _assertActionDropdownItemViewEntries(
+		DropdownItem dropdownItem) {
+
+		Assert.assertEquals(
+			_INVALID_DDM_FORM_INSTANCE, dropdownItem.get("disabled"));
+		Assert.assertEquals(
+			StringBundler.concat(
```

Comment:
As offline discussed, let's get rid of the URL matching. Just make sure that the other attributes are ok.

Reply:
Done!